substrate/frame/core-fellowship/src/migration.rs

```diff
@@ -0,0 +1,85 @@
+use super::*;
+use frame_support::{
+	pallet_prelude::*, storage_alias, traits::UncheckedOnRuntimeUpgrade, BoundedVec,
+};
+/// The log target of this pallet.
+pub const LOG_TARGET: &str = "runtime::core_fellowship";
+
+mod v0 {
+	use frame_system::pallet_prelude::BlockNumberFor;
+
+	use super::*;
+
+	#[derive(Encode, Decode, Eq, PartialEq, Clone, TypeInfo, MaxEncodedLen, RuntimeDebug)]
+	pub struct ParamsType<Balance, BlockNumber, const RANKS: usize> {
+		pub active_salary: [Balance; RANKS],
+		pub passive_salary: [Balance; RANKS],
+		pub demotion_period: [BlockNumber; RANKS],
+		pub min_promotion_period: [BlockNumber; RANKS],
+		pub offboard_timeout: BlockNumber,
+	}
+
+	impl<Balance: Default + Copy, BlockNumber: Default + Copy, const RANKS: usize> Default
+		for ParamsType<Balance, BlockNumber, RANKS>
+	{
+		fn default() -> Self {
+			Self {
+				active_salary: [Balance::default(); RANKS],
+				passive_salary: [Balance::default(); RANKS],
+				demotion_period: [BlockNumber::default(); RANKS],
+				min_promotion_period: [BlockNumber::default(); RANKS],
+				offboard_timeout: BlockNumber::default(),
+			}
+		}
+	}
+
+	/// Number of available ranks from old version.
+	pub(crate) const RANK_COUNT: usize = 9;
+
+	pub type ParamsOf<T, I> = ParamsType<<T as Config<I>>::Balance, BlockNumberFor<T>, RANK_COUNT>;
+
+	/// V0 type for [`crate::Params`].
+	#[storage_alias]
+	pub type Params<T: Config<I>, I: 'static> =
+		StorageValue<Pallet<T, I>, ParamsOf<T, I>, ValueQuery>;
+}
+
+mod v1 {
+	use super::*;
+
+	pub struct Migration<T, I = ()>(PhantomData<(T, I)>);
+	impl<T: Config<I>, I: 'static> UncheckedOnRuntimeUpgrade for Migration<T, I> {
+		fn on_runtime_upgrade() -> frame_support::weights::Weight {
```

Comment:
The `pre_updated` hook would be good to use here to check that the new number of ranks will not truncate.

Reply:
you mean the `post_upgrade`?

Reply:
Either would do. Just something to that nothing got/gets truncated.